### PR TITLE
Pullquote block: Fix citation markup in Editor

### PR DESCRIPTION
**Pull request by ndiego**
## What?
This PR changes the pullquote citation markup in the Editor from `<div>` to `<cite>` to improve consistency between the Editor and Frontend. The Frontend already uses `<cite>`.

## Why?
This bug appears to have been an oversight, the quote block already has this fix. This PR copies the quote block implementation exactly. 

## Testing Instructions
1. Add a pullquote block to a page/post. 
2. Confirm the citation markup in the Editor is `<cite>`, and so is the Frontend. 

Props to @mrleemon for the original PR: #38610

### CODE REVIEW

**Comment by carolinan** [comment]
Works well in the editors, thank you. The failing tests seems unrelated.

**Comment by ndiego** [comment]
Thanks for reviewing @carolinan, I'll give it a day or so and rebase to try and clear the failing tests. They do appear unrelated.

**Comment by skorasaurus** [comment]
Thanks @ndiego 

related issue: https://github.com/WordPress/gutenberg/issues/19847 

There is another PR for this already at https://github.com/WordPress/gutenberg/pull/38610 ; I didn't check whether the two approaches are the same. 


For what it's worth: 

This arose when I was investigating my own organization's internal use of the pullquote block a few months ago: 

The cite element may not be the most appropriate HTML element to be used because it is to only be used to refer to a [published work](https://html.spec.whatwg.org/#the-cite-element) and not to be used to refer to a person (which my org have been using it as such for some time) which states that "person's name is not the title of a work — even if people call that person a piece of work" 

there is an issue discussing the appropriate HTML element https://github.com/WordPress/gutenberg/issues/20606

I understand that the ship (of using the cite element) has already sort of sailed so but thought you and any other reviews should know.

**Comment by ndiego** [comment]
Thanks for calling this out @skorasaurus, I missed that other PR! This PR matches the current quote block implementation exactly. Understanding the potential concerns around the use of `<cite>`, I think we should ship this change (this PR or https://github.com/WordPress/gutenberg/pull/38610) to establish consistency especially since the pullquote block is already using `<cite>` on the frontend. A full overhaul of our use of `<cite>` could be its own PR 🤔

**Comment by ndiego** [comment]
All tests have passed, just waiting on approval

File: packages/block-library/src/pullquote/edit.js
Comment (aristath) [inline packages/block-library/src/pullquote/edit.js:73]: Shouldn't this be in the block's `style.scss` file?

**Review by aristath — APPROVED** [review]
LGTM 👍 
I only have a small nitpick about the `style` added, but it's not something that should block this PR, so I'll go ahead and pre-approve it 👍

File: packages/block-library/src/pullquote/edit.js
Comment (ndiego) [reply packages/block-library/src/pullquote/edit.js:73]: Thanks @aristath, I am going to merge this as is since it matches the implementation of the Quote block exactly, so at least there is consistency. I will prepare a follow-up PR that goes through both blocks (and any others I find) and move these inline styles to the respective `style.scss` files. There is a decent amount of them.

**Pull request merged**
